### ISSUE TITLE
GPUImageView output is black sometimes

### ISSUE THREAD

**Issue by beeradmoore** [BradLarson/GPUImage#1369]
Hey,
I have a filter chain which is currently set up in the following way.
GPUImagePicture (from camera or album) > Mirror Filter (see below) > GPUImageHSBFilter -> GPUImageView

My mirror filter code is as follows

```
varying highp vec2 textureCoordinate;

uniform sampler2D inputImageTexture;
uniform lowp float mirrorOffset;

void main()
{
    mediump vec2 p = textureCoordinate;

    if (p.x > 0.5)
    {
        p.x =  1.5 - p.x - mirrorOffset;
    }
    else
    {
        p.x = p.x - mirrorOffset + 0.5;
    }


    lowp vec4 outputColor = texture2D (inputImageTexture, p);
    gl_FragColor = outputColor;
}
```

The idea is that given a value with the setFloat for mirrorOffset I can mirror an image and shift it slightly.
I adjust mirrorOffset with a slider, on the slider value changed I update the value for my filter and then call processImage on my original GPUImagePicture.
I also have 3 different sliders for my hue, saturation and lightness. Anytime one of these is changed I call reset on the HSL filter and then set the hue, saturation and brightness and then of course processImage on my original GPUImagePicture.

The filter chain is set up correctly as far as I can tell as when I move any of the 4 sliders the output image is updated correctly. However sometimes the output image screen just goes black. By using the imageFromCurrentlyProcessedOutput method I am able to output the image from each filter and see that they are all correct. 

So why would link between the GPUImageHSBFilter and the GPUImageView somehow break? I know it is not directly the GPUImageHSBFilter or my mirror filter as I can them without the other and still cause the issue to happen.
I have even gone as far as on every slider increment I remove all targets from the GPUImageHSBFilter and then re-add the GPUImageView as the target before I call process image and it does not fix the problem.

I have correctly kept a reference to each one of my filters and the only thing I haven't tried which I am pretty sure is wrong, is to create my filters again on every value change of the sliders.

I think most of the devices this has been tested on are iOS7 devices but we have tested on a 4, 4S, 5 and 5S and we can make the issue occur on all of them. No memory warnings are issued either.

Any thoughts on this?

**Comment by mitchellporter** [comment]
@ytn3rd did you ever figure this out?

**Comment by ytn3rd** [comment]
I don't believe I ever did. after EDIT2 above issue seemed very rare. Eventually project was sold onto new developers so I didn't have to worry about it anymore.
